jetty-websocket/websocket-common/src/main/java/org/eclipse/jetty/websocket/common/message/MessageInputStream.java

```diff
@@ -121,7 +113,30 @@ public void appendFrame(ByteBuffer framePayload, boolean fin) throws IOException
     @Override
     public void close()
     {
-        SuspendToken resume = null;
+        synchronized (this)
+        {
+            if (state == State.CLOSED)
+                return;
+
+            boolean remainingContent = (state != State.COMPLETE) ||
+                (!buffers.isEmpty() && buffers.peek() != EOF) ||
+                (activeBuffer != null && activeBuffer.hasRemaining());
+
+            if (remainingContent)
+                LOG.warn("MessageInputStream closed without fully consuming content");
+
+            state = State.CLOSED;
+            buffers.clear();
+            buffers.add(EOF);
+        }
+    }
+
+    @Override
+    public void messageComplete()
```

Comment:
I don't like this method concept.
We already have the "boolean fin" on the `appendMessage`, why are we not using it?

Reply:
I don't know, this concept already existed I didn't create it. Seems a bit redundant, but this is the reason we don't use the boolean `fin` in `appendFrame`.

I think its better to move this logic into the `appendFrame` method to get the state change done in the same synchronized block.